lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -994,6 +994,27 @@ namespace Js
             scriptContext = nullptr;
         }
     }
+
+    void JavascriptLibraryBase::CallNativeHostPromiseRejectionTracker(Var promise, Var reason, bool handled)
+    {
+        if (this->nativeHostPromiseRejectionTracker != nullptr)
+        {
+            BEGIN_LEAVE_SCRIPT(scriptContext);
+            try
+            {
+                this->nativeHostPromiseRejectionTracker(promise, reason, handled, this->nativeHostPromiseRejectionTrackerState);
+            }
+            catch (...)
+            {
+                // Hosts are required not to pass exceptions back across the callback boundary. If
+                // this happens, it is a bug in the host, not something that we are expected to
+                // handle gracefully.
+                Js::Throw::FatalInternalError();
```

Comment:
I would create a new ErrorReason in ReportError.h so that these bugs would be easy to spot in Watson

Reply:
I agree. This was in the original implementation (I just moved the code), but I definitely want to update this.

Reply:
FWIW there is a precedent for this type of FatalError in functions like `JavascriptLibrary::EnqueueTask`

Reply:
When I wrote this I intentionally duplicated what was done in EnqueueTask as assumed it should be consistent.